pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -57,6 +69,41 @@ static __always_inline bool is_http2(const char* buf, __u32 buf_size) {
     return is_http2_preface(buf, buf_size) || is_http2_server_settings(buf, buf_size);
 }
 
+// The method checks if the given buffer starts is amqp message.
+// Ref: https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf
+static __always_inline bool is_amqp(const char* buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, AMQP_MIN_FRAME_LENGTH)
+
+    // New connection should start with protocol header of AMQP.
+    // Ref https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf.
+    if (is_amqp_protocol_header(buf, buf_size)) {
+        return true;
+    }
+
```

Comment:
Once we know it is not a start of AMQP, we should check the size fits to our needs.
since you are reading up to the position `10` in the array, you must check the buffer size is at least `11`
So add
```
#define AMQP_MIN_PAYLOAD_LENGTH 11
```

```suggestion
    if (is_amqp_protocol_header(buf, buf_size)) {
        return true;
    }

    if (buf_size < AMQP_MIN_PAYLOAD_LENGTH) {
        return false;
    }
```